src/components/raycaster.js

```diff
@@ -148,98 +151,84 @@ module.exports.Component = registerComponent('raycaster', {
-      copyArray(this.prevIntersectedEls, this.intersectedEls);
-
-      // Raycast.
-      this.updateOriginDirection();
-      rawIntersections = this.raycaster.intersectObjects(this.objects, data.recursive);
-
-      // Only keep intersections against objects that have a reference to an entity.
-      intersections.length = 0;
-      for (i = 0; i < rawIntersections.length; i++) {
-        intersection = rawIntersections[i];
-        // Don't intersect with own line.
-        if (data.showLine && intersection.object === el.getObject3D('line')) {
-          continue;
-        }
-        if (intersection.object.el) {
-          intersections.push(intersection);
-        }
-      }
+  tick: function (time) {
+    var clearedIntersectedEls = this.clearedIntersectedEls;
+    var el = this.el;
+    var data = this.data;
+    var i;
+    var intersectedEls = this.intersectedEls;
+    var intersection;
+    var intersections = this.intersections;
+    var lineLength;
+    var prevCheckTime = this.prevCheckTime;
+    var prevIntersectedEls = this.prevIntersectedEls;
+    var rawIntersections;
+
+    // Only check for intersection if interval time has passed.
+    if (prevCheckTime && (time - prevCheckTime < data.interval)) { return; }
+    // Update check time.
+    this.prevCheckTime = time;
+
+    // Refresh the object whitelist if needed.
+    if (this.dirty) { this.refreshObjects(); }
+
+    // Store old previously intersected entities.
+    copyArray(this.prevIntersectedEls, this.intersectedEls);
+
+    // Raycast.
+    this.updateOriginDirection();
+    rawIntersections = this.raycaster.intersectObjects(this.objects, data.recursive);
+
+    // Only keep intersections against objects that have a reference to an entity.
+    intersections.length = 0;
+    for (i = 0; i < rawIntersections.length; i++) {
+      intersection = rawIntersections[i];
+      // Don't intersect with own line.
+      if (data.showLine && intersection.object === el.getObject3D('line')) { continue; }
+      if (intersection.object.el) { intersections.push(intersection); }
+    }
 
-      // Update intersectedEls.
-      intersectedEls.length = intersections.length;
-      for (i = 0; i < intersections.length; i++) {
-        intersectedEls[i] = intersections[i].object.el;
-      }
+    // Update intersectedEls.
+    intersectedEls.length = intersections.length;
+    for (i = 0; i < intersections.length; i++) {
+      intersectedEls[i] = intersections[i].object.el;
+    }
 
-      // Emit intersected on intersected entity per intersected entity.
-      for (i = 0; i < intersections.length; i++) {
-        intersections[i].object.el.emit('raycaster-intersected', {
-          el: el,
-          intersection: intersections[i]
-        });
-      }
+    // Emit intersected on intersected entity per intersected entity.
+    for (i = 0; i < intersections.length; i++) {
+      intersections[i].object.el.emit('raycaster-intersected', {
```

Comment:
we can use `intersectionDetail` for this one too

Reply:
We probably need an `intersectedDetail` object

Reply:
This one is different because it's different data per intersected entity.